app/models/post.rb

```diff
@@ -1,3 +1,13 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    user.update(postsCounter: user.posts.count)
+  end
```

Comment:
[OPTIONAL] It is a good practice to use rails callbacks like `after_save` in scenarios like this. This way you can make your update_posts_counter private making it more secure.